transaction.go

```diff
@@ -141,30 +141,24 @@ func (t *Transaction) commit(done chan error) {
 }
 
 // marshalUpdates marshals the updates of a transaction
-func marshalUpdates(updates []Update) ([]byte, error) {
+func marshalUpdates(updates []Update) []byte {
 	buffer := new(bytes.Buffer)
-
 	for _, update := range updates {
 		// Marshal name
 		name := []byte(update.Name)
-		err1 := binary.Write(buffer, binary.LittleEndian, uint64(len(name)))
-		_, err2 := buffer.Write(name)
+		_ = binary.Write(buffer, binary.LittleEndian, uint64(len(name)))
+		_, _ = buffer.Write(name)
 
 		// Marshal version
 		version := []byte(update.Version)
-		err3 := binary.Write(buffer, binary.LittleEndian, uint64(len(version)))
-		_, err4 := buffer.Write(version)
+		_ = binary.Write(buffer, binary.LittleEndian, uint64(len(version)))
+		_, _ = buffer.Write(version)
 
 		// Append instructions
-		err5 := binary.Write(buffer, binary.LittleEndian, uint64(len(update.Instructions)))
-		_, err6 := buffer.Write(update.Instructions)
-
-		if err1 != nil || err2 != nil || err3 != nil || err4 != nil || err5 != nil || err6 != nil {
-			return nil, errors.Compose(errors.New("Failed to marshal updates"), err1, err2, err3, err4, err5, err6)
-		}
+		_ = binary.Write(buffer, binary.LittleEndian, uint64(len(update.Instructions)))
+		_, _ = buffer.Write(update.Instructions)
```

Comment:
after we merge this I'll do another optimization pass that writes directly into a preallocated `[]byte` :P